File: Repository/TicketRepository.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using DatabaseConnection;
using Microsoft.Extensions.Logging;
using Domain;

namespace projectC.Repository
{
    public class TicketRepository : GenericInterface<Ticket, int>
    {
        private SQLiteConnection connection;
        private static readonly ILogger<TicketRepository> logger = LoggerFactory.Create(builder =>
        {
            builder.AddConsole();
        }).CreateLogger<TicketRepository>();

        public TicketRepository()
        {
            this.connection = DBRepo.Connect();
        }

        public void Add(Ticket ticket)
        {
            string sql = "INSERT INTO tickets (ticketId, ticketPrice, nr_seats, matchId_fk) VALUES (@ticketId, @ticketPrice, @nrSeats, @matchId)";

            try
            {
                using (SQLiteCommand cmd = new SQLiteCommand(sql, connection))
                {
                    cmd.Parameters.AddWithValue("@ticketId", ticket.ticketId);
                    cmd.Parameters.AddWithValue("@ticketPrice", ticket.ticketPrice);
                    cmd.Parameters.AddWithValue("@nrSeats", ticket.nr_seats);
                    cmd.Parameters.AddWithValue("@matchId", ticket.matchId);

                    int rowsAffected = cmd.ExecuteNonQuery();
                    if (rowsAffected > 0)
                        logger.LogInformation("✅ Ticket added successfully.");
                    else
                        logger.LogWarning("⚠️ Ticket was not added.");
                }
            }
            catch (Exception ex)
            {
                logger.LogError("❌ Error inserting ticket: {0}", ex.Message);
            }
        }

        public void Remove(int ticketId)
        {
            string sql = "DELETE FROM tickets WHERE ticketId = @ticketId";

            try
            {
                using (SQLiteCommand cmd = new SQLiteCommand(sql, connection))
                {
                    cmd.Parameters.AddWithValue("@ticketId", ticketId);
                    int rowsAffected = cmd.ExecuteNonQuery();

                    if (rowsAffected > 0)
                        logger.LogInformation("✅ Ticket removed successfully.");
                    else
                        logger.LogWarning("⚠️ Ticket was not removed.");
                }
            }
            catch (Exception ex)
            {
                logger.LogError("❌ Error removing ticket: {0}", ex.Message);
            }
        }

        public Ticket GetById(int ticketId)
        {
            string sql = "SELECT * FROM tickets WHERE ticketId = @ticketId";

            try
            {
                using (SQLiteCommand cmd = new SQLiteCommand(sql, connection))
                {
                    cmd.Parameters.AddWithValue("@ticketId", ticketId);
                    using (SQLiteDataReader resultSet = cmd.ExecuteReader())
                    {
                        if (resultSet.Read())
                        {
                            return new Ticket(
                                resultSet.GetInt32(0), // ticketId
                                resultSet.GetDouble(1), // ticketPrice
                                resultSet.GetInt32(2), // nr_seats
                                resultSet.GetInt32(3)  // matchId_fk
                            );
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                logger.LogError("❌ Error getting ticket: {0}", ex.Message);
            }
            return null;
        }

        public void Update(int ticketId, Ticket ticket)
        {
            string sql = "UPDATE tickets SET ticketPrice = @ticketPrice, nr_seats = @nrSeats, matchId_fk = @matchId WHERE ticketId = @ticketId";

            try
            {
                using (SQLiteCommand cmd = new SQLiteCommand(sql, connection))
                {
                    cmd.Parameters.AddWithValue("@ticketPrice", ticket.ticketPrice);
                    cmd.Parameters.AddWithValue("@nrSeats", ticket.nr_seats);
                    cmd.Parameters.AddWithValue("@matchId", ticket.matchId);
                    cmd.Parameters.AddWithValue("@ticketId", ticket.ticketId);

                    int rowsAffected = cmd.ExecuteNonQuery();
                    if (rowsAffected > 0)
                        logger.LogInformation("✅ Ticket updated successfully.");
                    else
                        logger.LogWarning("⚠️ Ticket was not updated.");
                }
            }
            catch (Exception ex)
            {
                logger.LogError("❌ Error updating ticket: {0}", ex.Message);
            }
        }

        public List<Ticket> GetAll()
        {
            List<Ticket> tickets = new List<Ticket>();
            string sql = "SELECT * FROM tickets";

            try
            {
                using (SQLiteCommand cmd = new SQLiteCommand(sql, connection))
                {
                    using (SQLiteDataReader resultSet = cmd.ExecuteReader())
                    {
                        while (resultSet.Read())
                        {
                            Ticket ticket = new Ticket(
                                resultSet.GetInt32(0),
                                resultSet.GetDouble(1),
                                resultSet.GetInt32(2),
                                resultSet.GetInt32(3)
                            );
                            tickets.Add(ticket);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                logger.LogError("❌ Error retrieving tickets: {0}", ex.Message);
            }
            return tickets;
        }
    }
}

File: project-java-BogdanBala112/HomeAssignment1/src/main/java/Repository/TicketRepository.java
package Repository;

import Entities.Match;
import Entities.Ticket;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import javax.xml.transform.Result;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class TicketRepository implements GenericInterface<Ticket, Integer>{
    private static final Logger logger = LogManager.getLogger(MatchRepository.class);

    public Connection connection;

    public TicketRepository() {
        this.connection = DBAccess.connect();
    }

    @Override
    public void add(Ticket t) {
        String sql = "INSERT INTO tickets (ticketId,nr_seats, matchId_fk) VALUES (?, ?, ?)";

        try(PreparedStatement stmt = connection.prepareStatement(sql)){
            stmt.setInt(1,t.getticketId());
            stmt.setInt(2,t.getnr_seats());
            stmt.setInt(3,t.getmatchId());

            int rowsAffected = stmt.executeUpdate();

            if (rowsAffected > 0) {
                logger.info("Ticket added successfully");
            }
            else{
                logger.warn("Ticket not added");
            }
        }
        catch(SQLException e){
            logger.error("Error inserting ticket" + e.getMessage());
        }
    }

    @Override
    public void remove(Integer ticketId){
        String sql = "DELETE FROM tickets WHERE ticketId = ?";

        try(PreparedStatement stmt = connection.prepareStatement(sql)){
            stmt.setInt(1,ticketId);

            int rowsAffected = stmt.executeUpdate();

            if (rowsAffected > 0) {
                logger.info("Ticket removed successfully");
            }
            else{
                logger.warn("Ticket not removed");
            }
        }
        catch(SQLException e){
            logger.error("Error removing ticket" + e.getMessage());
        }
    }

    @Override
    public Ticket getById(Integer ticketId){
        String sql = "SELECT * FROM ticket WHERE ticketId = ?";

        try(PreparedStatement stmt = connection.prepareStatement(sql)){
            stmt.setInt(1,ticketId);
            ResultSet resultSet = stmt.executeQuery();

            if (resultSet.next()){
                return new Ticket(
                        resultSet.getInt("ticketId"),
                        resultSet.getInt("nr_seats"),
                        resultSet.getInt("matchId_fk")
                );
            }
        }
        catch(SQLException e){
            logger.error("Error getting ticket" + e.getMessage());
        }
        return null;
    }

    @Override
    public void update(Integer ticketId, Ticket t){
        String sql = "UPDATE tickets SET nr_seats = ?,matchId_fk = ? WHERE ticketId = ?";

        try(PreparedStatement stmt = connection.prepareStatement(sql)){
            stmt.setInt(1,t.getnr_seats());
            stmt.setInt(2,t.getmatchId());
            stmt.setInt(3,t.getticketId());

            int rowsAffected = stmt.executeUpdate();

            if (rowsAffected > 0) {
                logger.info("Ticket updated successfully");
            }
            else{
                logger.warn("Ticket not updated");
            }
        }
        catch(SQLException e){
            logger.error("Error inserting ticket" + e.getMessage());
        }
    }

    public List<Ticket> getAll(){
        String sql = "SELECT * FROM tickets";
        List<Ticket> tickets = new ArrayList<>();

        try(PreparedStatement stmt = connection.prepareStatement(sql)){
            ResultSet resultSet = stmt.executeQuery();

            while (resultSet.next()){
                tickets.add(new Ticket(
                        resultSet.getInt("ticketId"),
                        resultSet.getInt("nr_seats"),
                        resultSet.getInt("matchId_fk")
                ));
            }
        }
        catch(SQLException e){
            logger.error("Error getting tickets" + e.getMessage());
        }
        return tickets;
    }

}
